test/badges.test.ts

```diff
@@ -8,129 +8,153 @@ import { splitSignature, _TypedDataEncoder } from 'ethers/lib/utils'
+    const { badgesContract } = await loadFixture(deployContractFixture)
+
+    const deployedContractName = await badgesContract.name()
+    const deployedSymbolName = await badgesContract.symbol()
+    expect(deployedContractName).to.equal(name)
+    expect(deployedSymbolName).to.equal(symbol)
+
+    const provider = waffle.provider
+    const network = await provider.getNetwork()
+    const deployedToChainId = network.chainId
+    expect(deployedToChainId).to.equal(chainId)
+  })
+
+  it('should match off-chain hash to on-chain hash', async () => {
+    const { badgesContract, typedData } = await loadFixture(deployContractFixture)
+    const offChainHash = _TypedDataEncoder.hash(typedData.domain, typedData.types, typedData.value)
+
+    const onChainHash = await badgesContract.getHash(
+      typedData.value.active,
+      typedData.value.passive,
+      typedData.value.tokenURI
+    )
+
+    expect(offChainHash).to.equal(onChainHash)
+  })
+
+  it('should fail to mintWithAuthorizedBadge when using incorrect issuer address', async () => {
+    const { badgesContract, typedData, issuer, claimant } = await loadFixture(deployContractFixture)
+    const signature = await issuer._signTypedData(typedData.domain, typedData.types, typedData.value)
+    const { compact } = splitSignature(signature)
+    const randomWallet = Wallet.createRandom()
+
+    await expect(
+      badgesContract.connect(claimant).mintAuthorizedBadge(randomWallet.address, typedData.value.tokenURI, compact)
+    ).to.be.revertedWith('mintAuthorizedBadge: badge minting failed')
+  })
+
+  it('should fail to mintWithAuthorizedBadge when using incorrect token uri', async () => {
+    const { badgesContract, typedData, issuer, claimant } = await loadFixture(deployContractFixture)
+    const signature = await issuer._signTypedData(typedData.domain, typedData.types, typedData.value)
+    const { compact } = splitSignature(signature)
+
+    await expect(
+      badgesContract
+        .connect(claimant)
+        .mintAuthorizedBadge(typedData.value.passive, 'https://some-incorrect-uri.com', compact)
+    ).to.be.revertedWith('mintAuthorizedBadge: spec is not registered')
+  })
+
+  it('should fail to mintWithAuthorizedBadge when using invalid signature', async () => {
+    const { badgesContract, typedData, claimant } = await loadFixture(deployContractFixture)
+    const sig = { compact: 'junk conpact value' }
+    const sigAsBytes = ethers.utils.hexlify(ethers.utils.toUtf8Bytes(JSON.stringify(sig)))
+
+    await expect(
+      badgesContract
+        .connect(claimant)
+        .mintAuthorizedBadge(typedData.value.passive, typedData.value.tokenURI, sigAsBytes)
+    ).to.be.revertedWith('mintAuthorizedBadge: badge minting failed')
+  })
+
+  it('should fail to mintWithAuthorizedBadge when using unauthorized claimant', async () => {
+    const { badgesContract, typedData, issuer, badActor } = await loadFixture(deployContractFixture)
```

Comment:
Maybe it's better to rename `badActor` to `claimant2` or `randomSigner` ? Could be confusing to infer what a badActor could mean in general and also within this context..